### DISCUSSION TITLE
Simplify postgres jsonb result with select

### DISCUSSION THREAD

**Discussion by sehmaschine** [prisma/prisma#4423] [Q&A] [answered]
I'm aware that this would translate into a rather complicated SQL query, but I'm wondering if this is (or should be) possible ...

With my data table, I have a (Postgres) jsonb field `sections` which contains an array of data (lots of nested data due to PDF parsing). With the following example, `country` is a relation which can nicely be retrieved by selecting attributes. Is it possible to do the same with the json field? 

Something like this:

```ts
const data = await prisma.data.findMany({
    select: {
        sections: {
            select: {
                name: true,
                results: {
                    name: true
                }
            }
        },
        country: {
            select: {
                id: true,
                name: true
            }
        }
    }
})
```

**Comment by ryands17** [comment, marked as answer]
Hey @sehmaschine 👋 

This would qualify for a feature request :)

Currently we do not know the type of the Json stored as Postgres doesn't enforce the type and validity of what is stored there, so this will have to be implemented at the schema level first and then using `select` would be possible.

It would be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

**Reply by sehmaschine** [reply]
Thanks @ryands17, I'll create a feature request. But I do think that Postgres knows about the type ...

Here's what I get with one of my DB tables:

```
    Column     |           Type           | Collation | Nullable |                   Default
---------------+--------------------------+-----------+----------+---------------------------------------------
 id            | integer                  |           | not null | 
 sections      | jsonb                    |           |          |
 category      | character varying(30)    |           | not null |
 create_date   | timestamp with time zone |           | not null |
 update_date   | timestamp with time zone |           | not null |
```

**Reply by ryands17** [reply]
Postgres knows that it's `jsonb`, but it doesn't know the exact structure of the data that it will contain. It will allow you to store any kind of values there as long as its valid Json.

So using `select` is tricky without enforcing a type inside the Json i.e. what will be stored in that field.

**Reply by sehmaschine** [reply]
The data within `jsonb` could be unstructured, but Postgres still allows for querying it (with json-functions). With regards to my usecases, I could probably live with defining the structure (with the Prisma schema), but I doubt that this could be a general solution. My example could be translated to an SQL query without knowing the exact structure of the data.

**Reply by sehmaschine** [reply]
Actually, I'm not sure about my previous statement anymore. Maybe I'm only able to translate the example to an SQL query, because **I know** the structure. However, I'll create a feature request. Thanks @ryands17

**Reply by sehmaschine** [reply]
#4436

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
